docs/content/guides/dagster/local_to_production_tutorial.mdx

```diff
@@ -0,0 +1,395 @@
+        return item
+
+    def fetch_max_item_id(self) -> int:
+        return requests.get("https://hacker-news.firebaseio.com/v0/maxitem.json", timeout=5).json()
+
+@resource
+def hn_api_client():
+    return HNAPIClient()
+```
+
+Now we can add these assets to our repository and materialize them via Dagit as part of our local development workflow
+
+```python file=/guides/dagster/local_to_prod_tutorial/repository/repository_v1.py startafter=start endbefore=end
+# repository.py
+
+@repository
+def repo():
+    return [
+        *with_resources(
+            [items, comments, stories],
+            resource_defs={"hn_client": hn_api_client, "io_manager": snowflake_io_manager}
+        )
+    ]
+```
+
+When we materialize the assets in Dagit we can add configuration for the `snowflake_io_manager` in the launchpad or we can use the `configured` API and update our repository to look like this:
+
+```python file=/guides/dagster/local_to_prod_tutorial/repository/repository_v2.py startafter=start endbefore=end
+# repository.py
+
+@repository
+def repo():
+    resource_defs = {
+        "hn_client": hn_api_client,
+        "io_manager": snowflake_io_manager.configured(
+            {
+                "account": "abc1234.us-east-1",
+                "user": "me@company.com",
+                "password": "my_super_secret_password",
+                "database": "SANDBOX",
+                "schema": "JAMIE"
+            }
+        ),
+    }
+
+    return [*with_resources([items, comments, stories], resource_defs=resource_defs)]
+```
+
+Now we can materialize the assets the Dagit and ensure that the data appears in snowflake as we expect:
+
+<!-- \[Image: Screen Shot 2022-07-19 at 3.21.32 PM.png] -->
+
+<Note>
+
+While we define our assets as pandas DataFrames, the snowflake IO manager automatically translates the asset data to and from snowflake tables. The Python asset name determines the snowflake table name. In this case three tables will be created: ITEMS, COMMENTS, and STORIES
+
+</Note>
+
+### Part Two: Testing
+
+Let’s write some tests! We don’t want to rely on external calls to the Hacker News API during out unit tests, so the first thing we will need to do is write a version of the Hacker News API client that will serve consistent data.
+
+<Note>
+
+At Dagster we believe one of the main barriers to testing data applications is that business logic depends on external, heavy-weight dependencies that are difficult to run in a testing scenario. This is why we developed the concept of [resources](/concepts/resources).
+
+When testing an asset or op with an external dependency (either a resource or IO manager), we recommend using a lightweight version of the resource that implements all of the methods used in the asset or op. You can use a Dagster-provided resource (like the in memory IO manager), or write your own implementation.
+
+For more information and recommendations on testing, see our [testing documentation](/concepts/testing).
+
```

Comment:
```suggestion
At Dagster we believe one of the main barriers to testing data applications is that business logic depends on external, heavy-weight dependencies that are difficult to run in a testing scenario. This is why we developed the concept of [resources](/concepts/resources).

When testing an asset or op with an external dependency - like a resource or I/O manager - we recommend using a lightweight version of the resource that implements all of the methods used in the asset or op. You can use a Dagster-provided resource (like the in-memory I/O manager) or write your own implementation.

For more information and recommendations on testing, refer to the [testing documentation](/concepts/testing).

```

Suggesting some small tweaks to vary the sentence structure a bit.

Reply:
this note doesn't exist anymore, resolving